src/utils/uid.ts

```diff
@@ -0,0 +1,23 @@
+import { getCurrentInstance } from 'vue';
+
+import type { ComponentInternalInstance } from 'vue';
+
+let _uid = 0;
+const _map = new WeakMap<ComponentInternalInstance, number>();
```

Comment:
any reason why `_map` has an underscore ?
Maybe it could be renamed to `instanceMap`